tuning/sft_trainer.py

```diff
@@ -481,7 +515,11 @@ def main(**kwargs):  # pylint: disable=unused-argument
             fusedops_kernels_config,
             exp_metadata,
         ) = parse_arguments(parser, job_config)
-        logger.debug(
+
+        # Function to set log level for python native logger and transformers training logger
+        training_args, _ = set_log_level(training_args)
```

Comment:
Why are you not using the `logging.getLogger()` that you returned? Since you are using the root logger, running
```py
logging.warning("Warning")
# above is the same as below
root_logger = logging.getLogger()
root_logger.warning("Warning")
```
Running the above is the same. So if you are not using the getLogger, you shouldn't return it/use it.


But it looks like it is best practice to use getLogger from [python docs](https://docs.python.org/3/library/logging.html#logging.getLogger) and [python logging howto](https://docs.python.org/3/howto/logging.html).
So then instead of running `logging.warning()` you'd run `logger.warning()`. In which in the train() you would also call `logging.getLogger()`. It looks like its best practice to create module level loggers. So you could do `logging.getLogger("sft_trainer_main")` and logging.getLogger("sft_trainer_train")` and then the log messages should show like:

```sh
# named logger if you do getLogger("sft_trainer_train")
WARNING:sft_trainer_train:My warning

# root logger if you do getLogger()
WARNING:root:My warning
```

Reply:
In returned logger on line number 497, just to get logger in unit test cases functions. returning logger won't make sense in same file as we can set log level in same py file with returning logger.

But ya that's a good idea to use logging.getLogger. I will try and test it